lib/ui/channel_buffers.dart

```diff
@@ -0,0 +1,187 @@
+/// i.e. when a message handler is attached to the channel on the framework side.
+///
+/// Each channel has a finite buffer capacity and in a FIFO manner messages will
+/// be deleted if the capacity is exceeded.  The intention is that these buffers
+/// will be drained once a callback is setup on the BinaryMessenger in the
+/// Flutter framework.
+///
+/// Clients of Flutter shouldn't need to allocate their own ChannelBuffers
+/// and should only access this package's [channelBuffers] if they are writing
+/// their own custom [BinaryMessenger].
+class ChannelBuffers {
+  /// By default we store one message per channel.  There are tradeoffs associated
+  /// with any size.  The correct size should be chosen for the semantics of your
+  /// channel.
+  ///
+  /// Size 0 implies you want to ignore any message that gets sent before the engine
+  /// is ready (keeping in mind there is no way to know when the engine is ready).
+  ///
+  /// Size 1 implies that you only care about the most recent value.
+  ///
+  /// Size >1 means you want to process every single message and want to chose a
+  /// buffer size that will avoid any overflows.
+  static const int kDefaultBufferSize = 1;
+
+  final Map<String, _RingBuffer<_StoredMessage>> _messages =
+    <String, _RingBuffer<_StoredMessage>>{};
+
+  _RingBuffer<_StoredMessage> _makeRingBuffer(int size) {
+    final _RingBuffer<_StoredMessage> result = _RingBuffer<_StoredMessage>(size);
+    result.dropItemCallback = _onDropItem;
+    return result;
+  }
+
+  void _onDropItem(_StoredMessage message) {
+    message.callback(null);
+  }
+
+  /// Returns true on overflow.
+  bool push(String channel, ByteData data, PlatformMessageResponseCallback callback) {
+    _RingBuffer<_StoredMessage> queue = _messages[channel];
+    if (queue == null) {
+      queue = _makeRingBuffer(kDefaultBufferSize);
+      _messages[channel] = queue;
+    }
+    final bool result = queue.push(_StoredMessage(data, callback));
+    if (result) {
+      // TODO(aaclarke): Update this message to include instructions on how to resize
+      // the buffer once that is available to users.
+      _Logger._printString('Overflow on channel: $channel.  '
+                           'Messages on this channel are being discarded.  '
+                           'The engine may not be running or you need to adjust '
+                           'the buffer size if of the channel.');
+    }
+    return result;
+  }
+
+  /// Returns null on underflow.
+  _StoredMessage _pop(String channel) {
+    final _RingBuffer<_StoredMessage> queue = _messages[channel];
+    final _StoredMessage result = queue?.pop();
+    return result;
+  }
+
+  bool _isEmpty(String channel) {
+    final _RingBuffer<_StoredMessage> queue = _messages[channel];
+    return (queue == null) ? true : queue.isEmpty;
+  }
+
+  /// Changes the capacity of the queue associated with the given channel.
+  ///
+  /// This could result in the dropping of messages if the newSize is less
```

Comment:
nit `newSize`

Reply:
No article?  Done.

Reply:
ha, I'm sure there was a goomics episode about this but I can't find it anymore. 

I actually meant use back-ticks ` around method argument references in docs.